src/core/exec-credential.c

```diff
@@ -775,11 +773,11 @@ static int acquire_credentials(
                  * propagate a credential passed to us from further up. */
 
                 if (path_is_absolute(lc->path)) {
-                        sub_fd = open(lc->path, O_DIRECTORY|O_CLOEXEC|O_RDONLY);
```

Comment:
why is read only dropped here?

Reply:
If O_DIRECTORY is specified, the mode is effectively unused? I've removed O_RDONLY at several other places like this.